crates/biome_js_analyze/src/assist/source/organize_imports.rs

```diff
@@ -1,143 +1,328 @@
     ///
-    /// ```js
-    /// // group 1, only these two files will be sorted
-    /// import "../styles/reset.css";
-    /// import "../styles/layout.css";
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "groups": [
+    ///             ["@any/lib", "@any/lib/**", "!@any/lib/special", "!@any/lib/special/**"],
+    ///             "@/**"
+    ///         ]
+    ///     }
+    /// }
+    /// ```
     ///
-    /// // group 2, only this one is sorted
-    /// import { Grid } from "../components/Grid.jsx";
+    /// ```js,expect_diagnostic,use_options
+    /// import lib from "@any/lib";
+    /// import aliased from "@/alias";
+    /// import path from "@any/lib/special";
+    /// import test from "@any/lib/path";
     /// ```
     ///
-    /// ```js
-    /// // group 1, the polyfill/shim
-    /// import "../polyfills/array/flatMap";
+    /// Note that `@any/lib` matches `@any/lib` but not `@any/lib/**`.
+    /// Conversely, `@any/lib/subpath` matches `@any/lib/**`, but not `@any/lib`.
+    /// Thus you have to specify the two patterns if you want to accept all imports/exports that starts with `@any/lib`.
+    /// The prefix `!` indicates an exception.
+    /// For more details on the supported glob patterns, see the dedicated section.
+    ///
+    /// For now, it is not possible to mix predefined group and globs inside a list globs.
+    /// This is a limitation that we are working to remove.
     ///
-    /// // group 2, the files that require the polyfill/shim
-    /// import { functionThatUsesFlatMap } from "./utils.js";
+    /// The import organizer also allows you to indicate you want to separate two groups with a blank line using the predefined string `:BLANK-LINE:` like demonstrated in the following example.
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "groups": [
+    ///             ":NODE:",
+    ///             ":BLANK-LINE:"
+    ///             ["@any/lib", "@any/lib/**", "!@any/lib/special", "!@any/lib/special/**"],
+    ///             "@/**"
+    ///         ]
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic,use_options
+    /// import path from "node:path";
+    /// import lib from "@any/lib";
+    /// import test from "@any/lib/path";
+    /// import path from "@any/lib/special";
+    /// import aliased from "@/alias";
     /// ```
     ///
-    /// ## Side effect imports
+    /// Once divided in chunk and groups, imports and exports are sorted according to their source.
+    ///
+    /// ## Import and export sources
     ///
-    /// [Side effect imports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#forms_of_import_declarations) are import statements that usually don't import any name:
+    /// The source of an import or export is the string coming just after the `from` keyword.
```

Comment:
We used the word "source" many times , and the explanation comes only at this point of the docs. We should explain it at the very beginning. 

Also, I know you decided to use the word "source" because it might be more beginner-friendly, but I wonder if we should at least mention that it is *technically* called specifier.